File: vaultx-frontend/src/services/authService.js
import api from './api';

export const authService = {
  login: async (email, password, rememberMe) => {
    const response = await api.post('/auth/login', { email, password, rememberMe });
    return response.data;
  },
  
  register: async (userData) => {
    const response = await api.post('/auth/register', userData);
    return response.data;
  },

  sendEmailOtp: async (email) => {
    const response = await api.post('/auth/send-email-otp', { email });
    return response.data;
  },

  verifyEmailOtp: async (email, otp) => {
    const response = await api.post('/auth/verify-email-otp', { email, otp });
    return response.data;
  },

  sendMobileOtp: async (phoneNumber) => {
    const response = await api.post('/auth/send-mobile-otp', { phoneNumber });
    return response.data;
  },

  verifyMobileOtp: async (phoneNumber, otp) => {
    const response = await api.post('/auth/verify-mobile-otp', { phoneNumber, otp });
    return response.data;
  },

  sendRegistrationOtp: async (email, phoneNumber, username) => {
    const response = await api.post('/auth/send-registration-otp', { email, phoneNumber, username });
    return response.data;
  },

  registerWithOtp: async (userDataWithOtp) => {
    const response = await api.post('/auth/register-with-otp', userDataWithOtp);
    return response.data;
  },
  
  logout: async (refreshToken) => {
    const response = await api.post('/auth/logout', { refreshToken });
    return response.data;
  },
  
  logoutAll: async () => {
    const response = await api.post('/auth/logout-all');
    return response.data;
  },

  verifyEmail: async (token) => {
    const response = await api.get(`/auth/verify-email?token=${token}`);
    return response.data;
  },

  verifyOtp: async (phoneNumber, token) => {
    const response = await api.post('/auth/verify-otp', { phoneNumber, token });
    return response.data;
  },

  forgotPassword: async (email) => {
    const response = await api.post('/auth/forgot-password', { email });
    return response.data;
  },

  resetPassword: async (token, newPassword) => {
    const response = await api.post('/auth/reset-password', { token, newPassword });
    return response.data;
  },

  changePassword: async (currentPassword, newPassword) => {
    const response = await api.post('/auth/change-password', { currentPassword, newPassword });
    return response.data;
  }
};

File: vaultx-backend/src/main/java/com/vaultx/service/AuthService.java
package com.vaultx.service;

import com.vaultx.dto.auth.*;
import jakarta.servlet.http.HttpServletRequest;

public interface AuthService {
    String sendEmailOtp(SendEmailOtpRequest request);
    void verifyEmailOtp(VerifyEmailOtpRequest request);
    String sendMobileOtp(SendMobileOtpRequest request);
    void verifyMobileOtp(VerifyMobileOtpRequest request);
    void sendRegistrationOtp(SendOtpRequest request);
    void registerWithOtp(RegisterWithOtpRequest request);
    void register(RegisterRequest request);
    void verifyEmail(String token);
    void verifyOtp(VerifyOtpRequest request);
    JwtAuthenticationResponse login(LoginRequest request, HttpServletRequest httpRequest);
    JwtAuthenticationResponse walletLogin(WalletLoginRequest request, HttpServletRequest httpRequest);
    JwtAuthenticationResponse faceLogin(FaceLoginRequest request, HttpServletRequest httpRequest);
    JwtAuthenticationResponse refreshToken(TokenRefreshRequest request, HttpServletRequest httpRequest);
    void logout(String refreshToken, HttpServletRequest httpRequest);
    void logoutAllDevices(HttpServletRequest httpRequest);
    void forgotPassword(ForgotPasswordRequest request);
    void resetPassword(ResetPasswordRequest request);
    void changePassword(ChangePasswordRequest request, HttpServletRequest httpRequest);
}
